runtime/v2/shim/shim.go

```diff
@@ -271,11 +272,12 @@ func run(id string, initFunc Init, config Config) error {
 }
 
 // NewShimClient creates a new shim server client
-func NewShimClient(ctx context.Context, svc shimapi.TaskService, signals chan os.Signal) *Client {
+func NewShimClient(ctx context.Context, svc shimapi.TaskService, signals chan os.Signal, cancel func()) *Client {
```

Comment:
`context.CancelFunc` may look better than `func()`

Reply:
I don't have access to svc.cancel here, this is an interface.